From ajit.khaparde@emulex.com  Thu Jun 23 15:40:26 2011
From: Ajit Khaparde <ajit.khaparde@emulex.com>
Date: Tue, 19 Apr 2011 12:11:55 +0000
Subject: [PATCH] be2net: add code to display default value of tx rate for VFs
Git-commit: d0381c42aabdbd9402501d08ea44a89695ad58b4
Patch-mainline: v3.0-rc1
References: FATE#311448, bnc#697255

This change will allow the default value of tx rate to be displayed
when ip link show is called on a PF interface.

Signed-off-by: Ajit Khaparde <ajit.khaparde@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>

Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/benet/be_main.c |   25 +++++++++++++++++++------
 1 file changed, 19 insertions(+), 6 deletions(-)
--- a/drivers/net/benet/be_main.c
+++ b/drivers/net/benet/be_main.c
@@ -2252,12 +2252,6 @@ static int be_setup(struct be_adapter *a
 	if (status != 0)
 		goto rx_qs_destroy;
 
-	if (be_physfn(adapter) && adapter->sriov_enabled) {
-		status = be_vf_eth_addr_config(adapter);
-		if (status)
-			goto mcc_q_destroy;
-	}
-
 	adapter->link_speed = -1;
 
 	return 0;
@@ -2994,6 +2988,25 @@ static int __devinit be_probe(struct pci
 		goto unsetup;
 	netif_carrier_off(netdev);
 
+ 	if (be_physfn(adapter) && adapter->sriov_enabled) {
+		u8 mac_speed;
+		bool link_up;
+		u16 vf, lnk_speed;
+
+ 		status = be_vf_eth_addr_config(adapter);
+ 		if (status)
+ 			goto unreg_netdev;
+
+		for (vf = 0; vf < num_vfs; vf++) {
+			status = be_cmd_link_status_query(adapter, &link_up,
+					&mac_speed, &lnk_speed, vf + 1);
+			if (!status)
+				adapter->vf_cfg[vf].vf_tx_rate = lnk_speed * 10;
+			else
+				goto unreg_netdev;
+		}
+ 	}
+
 	dev_info(&pdev->dev, "%s port %d\n", nic_name(pdev), adapter->port_num);
 	schedule_delayed_work(&adapter->work, msecs_to_jiffies(100));
 	return 0;
